### PR TITLE
feat(pixel-spacing): enhance pixel spacing calculations with new measurement messages and validation functions

### PR DESCRIPTION
**Pull request by sedghi**
This pull request enhances the logic and robustness of pixel spacing calibration in radiographic imaging utilities, adding clearer handling for various calibration scenarios and significantly expanding the unit tests to cover edge cases. The main functional improvements include the introduction of standardized measurement messages, more precise branching for calibration types, and improved validation for pixel spacing values.

Key improvements and additions:

**Calibration Logic & Messaging Enhancements:**

* Introduced a `MeasurementMessages` object to standardize user-facing messages for different calibration outcomes, improving clarity and maintainability.
* Refactored the `calculateRadiographicPixelSpacing` function to:
  - Use new helper functions (`hasSpacing`, `isValidSpacing`) for safer pixel spacing validation.
  - Add more granular calibration type handling, including explicit messages for each scenario (e.g., user-calibrated, corrected at modality, corrected using ERMF, not calibrated, uncertain).
  - Ensure correct prioritization when both `FIDUCIAL` and `GEOMETRY` calibration types are present, and handle invalid or missing spacing data gracefully. [[1]](diffhunk://#diff-a1d75b850580a99c69238cc6723ebc50d2f56f238c0255814340a926985a48d1R93-R129) [[2]](diffhunk://#diff-a1d75b850580a99c69238cc6723ebc50d2f56f238c0255814340a926985a48d1R138-R190)

**Testing Improvements:**

* Greatly expanded the test suite for pixel spacing utilities:
  - Added comprehensive tests for all calibration scenarios, including edge cases such as missing or invalid spacing, illegal ERMF values, and both projection and non-projection SOPClassUIDs.
  - Used spies to verify warning/error logging behavior.
  - Ensured all new message and calibration branches are covered by assertions. [[1]](diffhunk://#diff-eaedaa905bf55807dd71203260c79f69ccde4033b6ed86f3575901fb99617c57R4-R32) [[2]](diffhunk://#diff-eaedaa905bf55807dd71203260c79f69ccde4033b6ed86f3575901fb99617c57L39-R62) [[3]](diffhunk://#diff-eaedaa905bf55807dd71203260c79f69ccde4033b6ed86f3575901fb99617c57R71-R289)

**Validation & Error Handling:**

* Improved validation for `PixelSpacing` and `ImagerPixelSpacing` to ensure only valid arrays of positive numbers are accepted, preventing subtle bugs and incorrect calibration results. [[1]](diffhunk://#diff-a1d75b850580a99c69238cc6723ebc50d2f56f238c0255814340a926985a48d1R48-R70) [[2]](diffhunk://#diff-a1d75b850580a99c69238cc6723ebc50d2f56f238c0255814340a926985a48d1R93-R129)
* Added explicit error logging for illegal ERMF values and ensured fallback logic is robust in these cases. [[1]](diffhunk://#diff-a1d75b850580a99c69238cc6723ebc50d2f56f238c0255814340a926985a48d1R138-R190) [[2]](diffhunk://#diff-eaedaa905bf55807dd71203260c79f69ccde4033b6ed86f3575901fb99617c57R71-R289)

These changes make the calibration logic more robust, user-friendly, and easier to maintain, while the improved test coverage helps ensure correctness and easier future refactoring.

**References:** [[1]](diffhunk://#diff-a1d75b850580a99c69238cc6723ebc50d2f56f238c0255814340a926985a48d1R48-R70) [[2]](diffhunk://#diff-a1d75b850580a99c69238cc6723ebc50d2f56f238c0255814340a926985a48d1R93-R129) [[3]](diffhunk://#diff-a1d75b850580a99c69238cc6723ebc50d2f56f238c0255814340a926985a48d1R138-R190) [[4]](diffhunk://#diff-eaedaa905bf55807dd71203260c79f69ccde4033b6ed86f3575901fb99617c57R4-R32) [[5]](diffhunk://#diff-eaedaa905bf55807dd71203260c79f69ccde4033b6ed86f3575901fb99617c57L39-R62) [[6]](diffhunk://#diff-eaedaa905bf55807dd71203260c79f69ccde4033b6ed86f3575901fb99617c57R71-R289)

### CODE REVIEW

File: packages/core/src/utilities/getPixelSpacingInformation.ts [hunk 1]
```diff
@@ -45,6 +45,29 @@ export function getERMF(instance) {
   }
 }
 
+const MeasurementMessages = Object.freeze({
```
Comment (wayfarer3130) [inline packages/core/src/utilities/getPixelSpacingInformation.ts]: The messages are a client side thing - these have different translations and are user facing.  It is ok in the examples, and I wouldn't mind having an examples support module containing shareable code for this type of thing, but it isn't a library sort of thing.

File: packages/core/src/utilities/getPixelSpacingInformation.ts [hunk 2]
```diff
@@ -45,6 +45,29 @@ export function getERMF(instance) {
...
+  CORRECTED_AT_MODALITY: 'Measurements corrected at modality.',
+  NOT_CORRECTED_AT_DETECTOR:
+    'Measurements not corrected. Measured size is at detector.',
+  CORRECTED_USING_ERMF: 'Measurements corrected using the ERMF.',
+  UNCERTAIN: 'Measurements are uncertain.',
+  USER_CALIBRATED: 'Measurements are user calibrated.',
+});
+
+function hasSpacing(spacing) {
+  return Array.isArray(spacing) && spacing.length === 2;
```
Comment (wayfarer3130) [inline packages/core/src/utilities/getPixelSpacingInformation.ts]: Some DICOM has just 1 spacing value, meaning it applies to both.  I don't think that is strictly valid.

File: packages/core/src/utilities/getPixelSpacingInformation.ts [hunk 3]
```diff
@@ -106,33 +135,59 @@ export function calculateRadiographicPixelSpacing(instance) {
...
+        Message: MeasurementMessages.CORRECTED_USING_ERMF,
       };
     }
+
     if (ermf === true) {
-      // PixelSpacing already updated/correct, don't tweak it
+      // PixelSpacing already updated/correct, don't tweak it again
       return {
-        PixelSpacing,
+        PixelSpacing: PixelSpacing || ImagerPixelSpacing,
```
Comment (wayfarer3130) [inline packages/core/src/utilities/getPixelSpacingInformation.ts]: Why did you add || ImagerPixelSpacing - the ermf===true means that PixelSpacing is already correct according to the ERMF.  The point of returning true is to avoid a recalculation that updates the floating values ever so slightly, and that only happens if PixelSpacing is defined.

File: packages/core/src/utilities/getPixelSpacingInformation.ts [hunk 4]
```diff
@@ -106,33 +135,59 @@ export function calculateRadiographicPixelSpacing(instance) {
...
         isProjection,
+        Message: MeasurementMessages.CORRECTED_USING_ERMF,
       };
     }
+
     if (ermf) {
       console.error('Illegal ERMF value:', ermf);
     }
+
+    if (hasValidImagerPixelSpacing && hasPixelSpacing) {
```
Comment (wayfarer3130) [inline packages/core/src/utilities/getPixelSpacingInformation.ts]: This is the comment for ermf===true, "Corrected at Modality"
Otherwise it is just projection spacing, and you should NOT indicate "Corrected at modality" when IPS===PS
since that is an invalid case - it means it is straight projection.

File: packages/core/src/utilities/getPixelSpacingInformation.ts (hunk 1 above)
Comment (wayfarer3130) [inline packages/core/src/utilities/getPixelSpacingInformation.ts]: I see no reason to freeze this, allowing it unfrozen allows the message to be updated.
